VideoLocker/src/main/java/org/edx/mobile/authentication/LoginAPI.java

```diff
@@ -52,43 +53,40 @@
     private final Gson gson;
 
     @Inject
-    public LoginAPI(@NonNull RestAdapter restAdapter,
+    public LoginAPI(@NonNull LoginService loginService,
                     @NonNull Config config,
                     @NonNull LoginPrefs loginPrefs,
                     @NonNull ISegment segment,
                     @NonNull NotificationDelegate notificationDelegate,
                     @NonNull Gson gson) {
+        this.loginService = loginService;
         this.config = config;
         this.loginPrefs = loginPrefs;
         this.segment = segment;
         this.notificationDelegate = notificationDelegate;
         this.gson = gson;
-        loginService = restAdapter.create(LoginService.class);
     }
 
     @NonNull
-    public AuthResponse getAccessToken(@NonNull String username,
-                                       @NonNull String password) throws HttpException {
+    public Response<AuthResponse> getAccessToken(@NonNull String username,
+                                       @NonNull String password) throws IOException {
         String grantType = "password";
         String clientID = config.getOAuthClientId();
-        return loginService.getAccessToken(grantType, clientID, username, password);
+        return loginService.getAccessToken(grantType, clientID, username, password).execute();
     }
 
     @NonNull
     public AuthResponse logInUsingEmail(@NonNull String email, @NonNull String password) throws Exception {
-        try {
-            final AuthResponse response = getAccessToken(email, password);
-            if (!response.isSuccess()) {
-                throw new AuthException(response.error);
-            }
-            finishLogIn(response, LoginPrefs.AuthBackend.PASSWORD, email.trim());
-            return response;
-        } catch (HttpResponseStatusException ex) {
-            if (ex.getStatusCode() >= 400 && ex.getStatusCode() < 500) {
-                throw new AuthException(ex);
-            }
-            throw ex;
+        final Response<AuthResponse> response = getAccessToken(email, password);
+        if (!response.isSuccessful()) {
+            throw new AuthException(response.message());
```

Comment:
If the response is an error, than the response message should be the error message in that case. The previous implementation just constructed an `AuthException` wrapper around the `HttpResponseStatusException`, which contained only the error code. This constructor is now unused, and could be removed I suppose.

The `AuthException` seems to be misused here though, as it's being thrown in the case of any HTTP error. But that's an existing issue which is out of the scope of this pull request.